pkgs/development/tools/build-managers/dub/default.nix

```diff
@@ -0,0 +1,35 @@
+{stdenv, fetchurl, curl, dmd, gcc, unzip}:
+
+stdenv.mkDerivation {
+  name = "dub-0.9.22";
+
+  src = fetchurl {
+    url = "https://github.com/rejectedsoftware/dub/archive/v0.9.22.tar.gz";
+    sha256 = "0vhn96ybbsfflldlbyc17rmwb7bz21slbm189k5glyfr9nnp4cir";
+  };
+
+  buildInputs = [ unzip curl ];
+
+  propagatedBuildInputs = [ gcc dmd ];
+
+  buildPhase = ''
+      # Avoid that the version file is overwritten
+      substituteInPlace build.sh \
+          --replace source/dub/version_.d /dev/null
+      ./build.sh
+  '';
+
+  installPhase = ''
+      mkdir $out
+      mkdir $out/bin
+      cp bin/dub $out/bin
+  '';
```

Comment:
Just tried this out and it did not work:

```
installing
install flags: install    
make: *** No rule to make target `install'.  Stop.
builder for ‘/nix/store/l3i79j7rn59jjk03di4hwpwwr8ididyi-dub-0.9.22.drv’ failed with exit code 2
```

Then I got curious and looked up the documentation, found this: http://nixos.org/nixpkgs/manual/#idm140737318835200

According to that it calls `make install` by default which will not work in this case since there is just no Makefile available here.

My conclusion is that writing down the `installPhase` a this little shell snippet is the right way to do it.